src/client/components/configuration/tree/tree.tsx

```diff
@@ -0,0 +1,66 @@
+import * as classnames from 'classnames'
+import { observer } from 'mobx-react'
+import * as React from 'react'
+
+import FileIcon from './file.svg'
+import FolderIconOpen from './folder-open.svg'
+import FolderIcon from './folder.svg'
+import * as style from './style.css'
+
+export interface Node {
+  label: string
+  expanded: boolean
+  leaf: boolean
+  selected: boolean
+  children?: Node[]
+}
+
+export interface TreeProps {
+  data: Node
+  level?: number
+  onClick(node: Node): void
+}
+
+@observer
+export class Tree extends React.Component<TreeProps, any> {
+  constructor(props: any, context: any) {
+    super(props, context)
+    this.onClick = this.onClick.bind(this)
+  }
+
+  public render(): JSX.Element {
+    const children = this.props.data.children || []
+    const hasChildren = children.length > 0
+    const level = this.props.level || 0
+    const classes = classnames(
+      style.treenode,
+      { [style['treenode--expanded']]: this.props.data.expanded },
+      { [style['treenode--selected']]: this.props.data.selected },
+    )
+    const headerInlineStyle = { paddingLeft: 8 + (level * 22) + 'px' }
+
+    return (
+      <div className={classes}>
+        <div className={style.treenode__header} onClick={this.onClick} style={headerInlineStyle}>
```

Comment:
I had that initially (see below) but changed to `paddingLeft` because I wanted the hover and selected highlights to be full width. I can switch back if necessary.

Using `ul`/`li`:

<img src="https://cloud.githubusercontent.com/assets/5924865/26393460/3bc887c0-40ad-11e7-8b74-49a59d422d78.png" width="320px">

Using `div` and `paddingLeft`:

<img src="https://cloud.githubusercontent.com/assets/5924865/26393652/d2dcdabc-40ad-11e7-8bc8-924fd19a4c0a.png" width="284px">